src/Microsoft.Diagnostics.EventFlow.Core/Implementations/DiagnosticPipeline.cs

```diff
@@ -20,10 +20,11 @@ namespace Microsoft.Diagnostics.EventFlow
     public class DiagnosticPipeline : IDisposable
     {
         private List<IDisposable> inputSubscriptions;
-        private IDisposable batcherTimer;
+        private Timer batcherTimer;
         private List<Task> outputCompletionTasks;
         private bool disposed;
         private bool disposeDependencies;
+        private object disposalLock;
```

Comment:
This is more of a batcherTimerDisposalLock than a general disposalLock, can we rename it to make it explicit?

Reply:
Good suggestion, thanks!